File: cpp/p1.cpp
#include <iostream>
#include <unordered_set>

using std::string;

class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        std::unordered_set<char> uset{};
        int i=0, j=0, result=0;
        if(s.length()<=0){
            return 0;
        }
        while(i<s.length()&&j<s.length()){
            if(uset.find(s[j])!=uset.end()){
                uset.erase(s[i]);
                i++;
            } else {
                uset.insert(s[j]);
                j++;
                result = result>j-i?result:j-i;
            }
        }

        return result;

    }
};

int main(int, char**){
    std::cout << "Hello, from leetcode!\n";
}

File: python/p1.py

class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        i =0, j = 0
        sub_set = set()
        sub_len = 0
        for i in range(0,len(s)):
            while j < len(s):
                if s[j] not in sub_set:
                    sub_set.add(s[j])
                    sub_len = j-i+1 if sub_len < j-i+1 else sub_len
                    j = j+1
                else:
                    sub_set.remove(s[i])
                    break
        return sub_len

        